src/main/java/org/olf/folio/order/OrderImport.java

```diff
@@ -464,10 +461,57 @@ public  JSONArray  upload(String fileName) throws IOException, InterruptedExcept
+                // TOSO: determine how this will be added to inventory. just grab the value for now
+                DataField nineSevenSix = (DataField) record.getVariableField("976");
+                String barcode = marcUtils.getBarcode(nineSevenSix);
+                if (StringUtils.isNotEmpty(barcode)) {
+                    logger.debug("get holdings for instanceId: "+ instanceId);
+                    String holdingsResponse = apiService.callApiGet(baseOkapEndpoint + "holdings-storage/holdings?query=(instanceId==" + instanceId + ")", token);
+                    JSONObject holdingsObject = new JSONObject(holdingsResponse);
+                    logger.debug(holdingsObject.toString(3));
+                    JSONObject holdingsAsJson = new JSONObject(holdingsResponse);
+                    
+                    // get holdings array
+                    JSONArray holdingsArray = holdingsAsJson.getJSONArray("holdingsRecords");
+                    logger.debug("holdingsArray size: "+ holdingsArray.length());
+                    Iterator  holdingsIter = holdingsArray.iterator();
+                    while (holdingsIter.hasNext() ) {
+                        JSONObject holdingsRecord = (JSONObject) holdingsIter.next();
+                        logger.debug("holdings record");
+                        logger.debug(holdingsRecord.toString(3));
+                        String holdingsId = holdingsRecord.getString("id"); 
+                        
+                        logger.debug("get inventory items with holdingsId: "+ holdingsId);
+                        String itemsResponse = apiService.callApiGet(baseOkapEndpoint + "inventory/items?query=(holdingsRecordId=" + holdingsId + ")", token);
+                        JSONObject itemsObject = new JSONObject(itemsResponse);
+                        logger.debug(itemsObject.toString(3));
+                        
+                        JSONArray itemsArray = itemsObject.getJSONArray("items");
+                         
+                        logger.debug("itemsArray size: "+ itemsArray.length());
+                        Iterator itemsIter = itemsArray.iterator();
+                        while (itemsIter.hasNext()) {
+                            JSONObject itemRecord = (JSONObject) itemsIter.next();
+                            itemRecord.put("barcode", barcode);
+                            
+                            String itemId = itemRecord.getString("id");
+                            logger.debug("item record: "+ itemId);
+                            logger.debug(itemRecord.toString(3));
+                            // PUT the item
+                            if (StringUtils.isNotEmpty(itemId)) {
+                                String itemPutResponse = apiService.callApiPut(baseOkapEndpoint + "inventory/items/" + itemId,  itemRecord, token);
+                                
+                            }
+                        }
```

Comment:
We definitely don't want to iterate through all the items for a holding and add the same barcode to each.

Somewhat serendipitously, I learned today that [FOLIO supports barcode validation and rejects duplicate barcodes](https://folio-project.slack.com/archives/CFQU1MF61/p1624986390202400), which is exactly what you're running up against here.

If you make the `PUT` request to Okapi directly, you can see as much from the response body:
```
Barcode must be unique, 31924129539775 is already assigned to another item
```

<img width="1440" alt="Screen Shot 2021-06-29 at 3 34 55 PM" src="https://user-images.githubusercontent.com/409842/123861535-4b7cf680-d8f5-11eb-9427-d2cbfa0a8727.png">


We know that submitting the same MARC file to the app multiple times will result in duplicate orders and Inventory items. LTS Acquisitions knows this as well and will do their best to avoid submitting the same file more than once.

That being said, at the very least I suggest we: 
1. Filter out items that already have barcodes
   - either in the initial `GET` to `/inventory/items` appending `not barcode=""` to the `query` URL param:
     ```
     query=(holdingsRecordId=="{holdingsId}" not barcode="")
     ```
   - or filter the response of the existing `GET` in Java
1. Add the barcode from the incoming `976$p` to the first item in the filtered array only